virttest/test_setup.py

```diff
@@ -97,6 +100,90 @@ class PolkitConfigCleanupError(PolkitConfigError):
     pass
 
 
+class Setuper(object):
+
+    """
+    Virtual base abstraction of setuper.
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, test, params, env):
+        """
+        Initializes the setuper.
+
+        :param test: VirtTest instance.
+        :param params: Dictionary with the test parameters.
+        :param env: Dictionary with test environment.
+        """
+        self.test = test
+        self.params = params
+        self.env = env
+
+    @abstractmethod
+    def setup(self):
+        """Setup procedure."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def cleanup(self):
+        """Cleanup procedure."""
+        raise NotImplementedError
+
+
+class SetuperMgr(object):
```

Comment:
Is there a reason to use shortcuts? It decreases the code readability. I'd suggest `SetupManager` (`Setuper` makes sense when it stands alone, but together with `Manager` it sounds strange).

Reply:
I just don't know how to name it, ok, will use `SetupManager`.